build-logic/testing-convention/src/main/kotlin/convention.test-fixtures.gradle.kts

```diff
@@ -16,3 +14,7 @@ configure<KotlinJvmProjectExtension> {
     target.compilations.getByName("testFixtures")
         .associateWith(target.compilations.getByName("main"))
 }
+
+val javaComponent = components["java"] as AdhocComponentWithVariants
+javaComponent.withVariantsFromConfiguration(configurations["testFixturesApiElements"]) { skip() }
+javaComponent.withVariantsFromConfiguration(configurations["testFixturesRuntimeElements"]) { skip() }
```

Comment:
How about `configurations.matching { name.startsWith("testFixtures") }` to find by prefix?

Reply:
`Variant for configuration testFixturesAnnotationProcessor does not exist in component java`

don't know how to associate cofigurations which is created by test-fixtures plugin and those that belongs to java software component

added some comments